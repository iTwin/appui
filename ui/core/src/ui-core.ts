/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
export * from "./ui-core/UiCore";
export * from "./ui-core/utils/Props";

export * from "./ui-core/base/UiEvent";
export * from "./ui-core/base/Div";
export * from "./ui-core/base/Centered";
export * from "./ui-core/base/FillCentered";
export * from "./ui-core/base/ScrollView";
export * from "./ui-core/base/FlexWrapContainer";

export * from "./ui-core/checklistbox/CheckListBox";

export * from "./ui-core/contextmenu/ContextMenu";

export * from "./ui-core/cube/Cube";

export * from "./ui-core/dialog/Dialog";

export * from "./ui-core/elementseparator/ElementSeparator";

export * from "./ui-core/enums/Alignment";
export * from "./ui-core/enums/CheckBoxState";
export * from "./ui-core/enums/DateFormat";
export * from "./ui-core/enums/Orientation";
export * from "./ui-core/enums/SortDirection";
export * from "./ui-core/enums/TimeFormat";

export * from "./ui-core/expandable/ExpandableList";
export * from "./ui-core/expandable/ExpandableBlock";

export * from "./ui-core/hocs/withIsPressed";
export * from "./ui-core/hocs/withOnOutsideClick";
export * from "./ui-core/hocs/withTimeout";

export * from "./ui-core/icons/SvgPath";
export * from "./ui-core/icons/SvgSprite";
export * from "./ui-core/icons/WebFontIcon";

export * from "./ui-core/messagebox/MessageBox";

export * from "./ui-core/popup/Popup";

export * from "./ui-core/radialmenu/RadialMenu";
export * from "./ui-core/radialmenu/Annulus";

export * from "./ui-core/imagecheckbox/ImageCheckBox";

export * from "./ui-core/searchbox/SearchBox";

export * from "./ui-core/splitbutton/SplitButton";

export * from "./ui-core/loading/LoadingBar";
export * from "./ui-core/loading/LoadingPrompt";
export * from "./ui-core/loading/LoadingSpinner";
export * from "./ui-core/loading/LoadingStatus";
export * from "./ui-core/loading/Spinner";

export * from "./ui-core/toggle/Toggle";

export { ExpansionToggle, ExpansionToggleProps } from "./ui-core/tree/ExpansionToggle";
export { TreeBranch, TreeBranchProps } from "./ui-core/tree/Branch";
export { TreeNode, TreeNodeProps, NodeCheckboxProps, NodeCheckboxRenderer, NodeCheckboxRenderProps } from "./ui-core/tree/Node";
export { Tree, TreeProps } from "./ui-core/tree/Tree";
export { TreeNodePlaceholder, TreeNodePlaceholderProps } from "./ui-core/tree/Placeholder";

export * from "./ui-core/uisettings/UiSettings";
export * from "./ui-core/uisettings/LocalUiSettings";

export * from "./ui-core/utils/Timer";
export * from "./ui-core/utils/getDisplayName";
export * from "./ui-core/utils/getUserColor";
export * from "./ui-core/utils/shallowDiffers";
export * from "./ui-core/utils/typeUtils";
export * from "./ui-core/utils/isPromiseLike";

export * from "./ui-core/button/Button";
export * from "./ui-core/button/UnderlinedButton";

export * from "./ui-core/inputs/checkbox/Checkbox";
export * from "./ui-core/inputs/Input";
export * from "./ui-core/inputs/InputStatus";
export * from "./ui-core/inputs/LabeledComponentProps";
export * from "./ui-core/inputs/LabeledInput";
export * from "./ui-core/inputs/LabeledSelect";
export * from "./ui-core/inputs/LabeledTextarea";
export * from "./ui-core/inputs/Radio";
export * from "./ui-core/inputs/Select";
export * from "./ui-core/inputs/Textarea";

export * from "./ui-core/tabs/HorizontalTabs";

export * from "./ui-core/text/BodyText";
export * from "./ui-core/text/Headline";
export * from "./ui-core/text/LeadingText";
export * from "./ui-core/text/LeadingText2";
export * from "./ui-core/text/MutedText";
export * from "./ui-core/text/SmallText";
export * from "./ui-core/text/Subheading";
export * from "./ui-core/text/Subheading2";
export * from "./ui-core/text/StyledText";
export * from "./ui-core/text/TextProps";
export * from "./ui-core/text/Title";
export * from "./ui-core/text/Title2";

// Set the version number so it can be found at runtime. BUILD_SEMVER is replaced at build time by the webpack DefinePlugin.
declare var BUILD_SEMVER: string;
// istanbul ignore if
if ((typeof (BUILD_SEMVER) !== "undefined") && (typeof window !== "undefined") && window) {
  if (!(window as any).iModelJsVersions)
    (window as any).iModelJsVersions = new Map<string, string>();
  (window as any).iModelJsVersions.set("ui-core", BUILD_SEMVER);
}

/** @docs-package-description
 * The ui-core package contains general purpose React components, such as Dialog, MessageBox, SearchBox, RadialMenu and SplitButton.
 * For more information, see [learning about ui-core]($docs/learning/core/index.md).
 */
/**
 * @docs-group-description Base
 * Low-level classes and components for building application UI.
 */
/**
 * @docs-group-description Button
 * Classes for working with various Buttons.
 */
/**
 * @docs-group-description CheckListBox
 * Classes for working with a Check listbox.
 */
/**
 * @docs-group-description Common
 * Common classes used across various UI components.
 */
/**
 * @docs-group-description ContextMenu
 * Classes for working with a Context Menu.
 */
/**
 * @docs-group-description Cube
 * Component for 3D Cube.
 */
/**
 * @docs-group-description Dialog
 * Classes for working with a Dialog or MessageBox.
 */
/**
 * @docs-group-description ElementSeparator
 * Classes for working with a ElementSeparator.
 */
/**
 * @docs-group-description Expandable
 * Classes for working with a ExpandableBlock or ExpandableList.
 */
/**
 * @docs-group-description Icon
 * Component that renders ui-core icon when given an icon name.
 */
/**
 * @docs-group-description Inputs
 * Classes for working with input controls, such as Input, Radio, Checkbox, Select and Toggle.
 */
/**
 * @docs-group-description Loading
 * Classes for working with Loading spinner and progress indicator.
 */
/**
 * @docs-group-description Popup
 * Classes for working with a Popup.
 */
/**
 * @docs-group-description RadialMenu
 * Classes for working with a RadialMenu.
 */
/**
 * @docs-group-description SearchBox
 * Classes for working with a SearchBox.
 */
/**
 * @docs-group-description SplitButton
 * Classes for working with a SplitButton.
 */
/**
 * @docs-group-description Tabs
 * Classes for working with horizontal tabs.
 */
/**
 * @docs-group-description Text
 * Classes for working with different text controls.
 */
/**
 * @docs-group-description Tree
 * Presentation React components for working with a Tree.
 */
/**
 * @docs-group-description UiSettings
 * Interfaces for working with persistent UI settings.
 */
/**
 * @docs-group-description Utilities
 * Various utility classes for working with a UI.
 */
