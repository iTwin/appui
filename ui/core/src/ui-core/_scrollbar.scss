/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "./colors";

$uicore-scrollbar-width: 10px;
$uicore-scrollbar-padding: 2px;

@mixin uicore-scrollbar($color: $background-4, $width: $uicore-scrollbar-width, $padding: $uicore-scrollbar-padding) {
  &::-webkit-scrollbar {
    width: $width;
    height: $width;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: $width;
    background-color: $color;
    background-clip: content-box;
    border-style: solid;
    border-color: transparent;
    border-width: $padding;
  }

  &::-webkit-scrollbar-corner {
    background-color: transparent;
  }

  &::-webkit-scrollbar-track-piece:corner-present {
    margin-bottom: -2 * $padding;
    margin-right: -2 * $padding;
  }
}

@mixin uicore-hidden-scrollbar {
  &::-webkit-scrollbar {
    width: 0;
    height: 0;
  }
}

@mixin uicore-touch-scrolling {
  @supports (-webkit-overflow-scrolling: touch) {
    overflow-y: scroll; /* overflow-y has to be scroll, not auto */
    -webkit-overflow-scrolling: touch;
  }
}
