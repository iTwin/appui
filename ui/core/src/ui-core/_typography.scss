/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
$display2-weight: normal;
$display2-size: 256%;
$display2-opacity: 0.88;

$display1-weight: normal;
$display1-size: 228%;
$display1-opacity: 0.88;

$headline-weight: normal;
$headline-size: 170%;
$headline-opacity: 1.00;

$title-weight: normal;
$title-size: 114%;
$title-opacity: 1.00;

$title-inactive-weight: lighter;
$title-inactive-size: 114%;
$title-inactive-opacity: 1.00;

$body-weight: lighter;
$body-size: 100%;
$body-opacity: 0.88;

$body1-weight: normal;
$body1-size: 100%;
$body1-opacity: 0.88;

$body2-weight: bolder;
$body2-size: 100%;
$body2-opacity: 0.88;

$body-disabled-weight: lighter;
$body-disabled-style: italic;
$body-disabled-size: 100%;
$body-disabled-opacity: 0.66;

$caption-weight: lighter;
$caption-size: 86%;
$caption-opacity: 0.88;

$caption1-weight: lighter;
$caption1-style: italic;
$caption1-size: 86%;
$caption1-opacity: 0.88;

@mixin title($rgb: unset) {
  font-weight: $title-weight;
  font-size: $title-size;

  @if $rgb != unset {
    color: rgba($color, $title-opacity);
  }
}

@mixin body($rgb: unset) {
  font-weight: $body-weight;
  font-size: $body-size;

  @if $rgb != unset {
    color: rgba($color, $body-opacity);
  }
}

@mixin body1($rgb: unset) {
  font-weight: $body1-weight;
  font-size: $body1-size;

  @if $rgb != unset {
    color: rgba($color, $body1-opacity);
  }
}