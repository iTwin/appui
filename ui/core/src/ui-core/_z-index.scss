/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "_helpers";

// Order matters, last is top-most on screen.
$uicore-internal-z-index-layers: content, zones, dragged-widget, zone-target, zone-outline, backstage-overlay, backstage, expandable, toolbar-histories, toolbar-panels, status-popover, status-message, dialog, tooltip;

$uicore-z-index-step: 1000;

@function uicore-internal-get-next-z-index-step($counter) {
  @return $counter + $uicore-z-index-step;
}

$uicore-internal-z-index-layer-map: uicore-map-from-array-with-result-arg($uicore-internal-z-index-layers, get-function(uicore-internal-get-next-z-index-step), 0);

// Use this to get z-index value
@function uicore-get-z-index($layer) {
  @if map_has_key($uicore-internal-z-index-layer-map, $layer) {
    @return map-get($uicore-internal-z-index-layer-map, $layer);
  }
  @else {
    @error "Failed to get z-index for $layer: #{$layer}";
  }
}

// Use this to add z-index property
@mixin uicore-z-index($layer) {
  z-index: uicore-get-z-index($layer);
}
