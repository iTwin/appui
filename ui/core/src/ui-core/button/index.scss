/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "./blue";
@import "./blue-large";
@import "./button";
@import "./disabled";
@import "./disabled-large";
@import "./hollow";
@import "./hollow-large";
@import "./primary";
@import "./primary-large";
