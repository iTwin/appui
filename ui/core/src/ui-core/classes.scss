/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./style/index";

@import "./base/classes";
@import "./inputs/classes";
@import "./text/classes";
@import "./button/classes";
@import "./expandable/classes";
@import "./icons/classes";
@import "./tabs/classes";
@import "./tiles/classes";
