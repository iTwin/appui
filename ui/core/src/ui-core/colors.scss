/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "./style/themecolors";

$background-1: var(--buic-background-1);  // #f9f9f9;
$background-2: var(--buic-background-2);  // #eef0f3;
$background-3: var(--buic-background-3);  // #dce0e3;
$background-4: var(--buic-background-4);  // #85929d;
$background-5: var(--buic-background-5);  // #4d575f;

$primary-base: $buic-foreground-primary; // #008be1;
$primary-tone: $buic-foreground-primary-tone; // #0072b8;
$primary-tint: $buic-foreground-primary-tint; // #a6d7f5;
$primary-hover: $buic-row-hover;  // #c9e9fc;

$accent-base: $buic-foreground-success; // #56aa1c;
$accent-tint: $buic-foreground-success-tint; // #c4e1b0;

$alert-base: $buic-foreground-alert; // #d30a0a;
$alert-tint: $buic-foreground-alert-tint; // #f0a9a9;

$text-dark: $buic-foreground-body;            // #000000;
$text-light: $buic-foreground-body-reverse;   // #ffffff;
