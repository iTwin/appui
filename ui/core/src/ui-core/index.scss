/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./style/index";

@import "./_scrollbar";
@import "./_z-index";

@import "./base/index";
@import "./inputs/index";
@import "./button/index";
@import "./expandable/index";
@import "./icons/index";
@import "./loading/index";
@import "./tabs/index";
@import "./text/index";
@import "./tiles/index";
