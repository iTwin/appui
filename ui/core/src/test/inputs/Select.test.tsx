/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import * as React from "react";
import { Select } from "../../ui-core";

describe("<Select />", () => {
  it("should render", () => {
    mount(<Select options={[]} />);
  });

  it("renders correctly", () => {
    shallow(<Select options={[]} />).should.matchSnapshot();
  });

  it("renders status correctly", () => {
    shallow(<Select options={[]} />).should.matchSnapshot();
  });

  it("renders placeholder correctly", () => {
    shallow(<Select options={[]} placeholder="test" />).should.matchSnapshot();
  });

  it("renders array options correctly", () => {
    shallow(<Select options={["Option 1", "Option 2", "Option 3"]} />).should.matchSnapshot();
  });

  it("renders object options correctly", () => {
    shallow(<Select options={{ option1: "Option 1", option2: "Option 2", option3: "Option3" }} />).should.matchSnapshot();
  });
});
