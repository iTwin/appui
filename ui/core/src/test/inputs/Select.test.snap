// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders array options correctly 1`] = `
<select
  className="uicore-inputs-select"
  defaultValue=""
  options={
    Array [
      "Option 1",
      "Option 2",
      "Option 3",
    ]
  }
>
  <option
    key="0"
    value="Option 1"
  >
    Option 1
  </option>
  <option
    key="1"
    value="Option 2"
  >
    Option 2
  </option>
  <option
    key="2"
    value="Option 3"
  >
    Option 3
  </option>
</select>
`;

exports[`<Select /> renders correctly 1`] = `
<select
  className="uicore-inputs-select"
  defaultValue=""
  options={Array []}
/>
`;

exports[`<Select /> renders defaultValue correctly 1`] = `
<select
  className="uicore-inputs-select"
  defaultValue="test"
  options={Array []}
/>
`;

exports[`<Select /> renders object options correctly 1`] = `
<select
  className="uicore-inputs-select"
  defaultValue=""
  options={
    Object {
      "option1": "Option 1",
      "option2": "Option 2",
      "option3": "Option3",
    }
  }
>
  <option
    key="option1"
    value="option1"
  >
    Option 1
  </option>
  <option
    key="option2"
    value="option2"
  >
    Option 2
  </option>
  <option
    key="option3"
    value="option3"
  >
    Option3
  </option>
</select>
`;

exports[`<Select /> renders placeholder correctly 1`] = `
<select
  className="uicore-inputs-select"
  defaultValue=""
  options={Array []}
  placeholder="test"
  required={true}
>
  <option
    className="placeholder"
    disabled={true}
    key=""
    value=""
  >
    test
  </option>
</select>
`;

exports[`<Select /> renders status correctly 1`] = `
<select
  className="uicore-inputs-select"
  defaultValue=""
  options={Array []}
/>
`;

exports[`<Select /> renders value correctly 1`] = `
<select
  className="uicore-inputs-select"
  defaultValue="test"
  options={Array []}
  value="test"
/>
`;
